packages/core/utils/lib/async.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const pMap = require('p-map');
+const { curry } = require('lodash/fp');
+
+function pipeAsync(...methods) {
+  return async (data) => {
+    let res = data;
+
+    for (const method of methods) {
+      res = await method(res);
+    }
+
+    return res;
+  };
+}
+
+/**
+ * @type { import('./async').MapAsync }
+ */
+const mapAsync = curry(pMap);
+
+/**
+ * @type { import('./async').reduceAsync }
+ */
+function reduceAsync(promiseArray) {
+  return (iteratee, initialValue) =>
+    promiseArray.reduce(async (previousPromise, currentValue, index) => {
+      const previousValue = await previousPromise;
+      return iteratee(previousValue, await currentValue, index);
+    }, Promise.resolve(initialValue));
```

Comment:
Looking at the code again I am concerned as it launches in parallel all the promises, which could be a lot and is not controlled. They are all launched in parallel but each one is waiting for the previous one so the parallel launch does improve performance.

What do you think of this instead?:
```js
const reduceAsync = curry(async (iteratee, initialValue, array) => {
  let acc = initialValue;
  for (let i = 0; i < array.length; i += 1) {
    acc = await iteratee(acc, array[i], i);
  }
  return acc;
});
```
I find it also more easy to understand